src/company_cmd.cpp

```diff
@@ -870,12 +867,16 @@ CommandCost CmdCompanyCtrl(TileIndex tile, DoCommandFlag flags, uint32 p1, uint3
 			break;
 		}
 
-		case CCA_NEW_AI: { // Make a new AI company
+		case CCA_NEW_AI: { // Make new AI companies
 			if (!(flags & DC_EXEC)) return CommandCost();
 
 			if (company_id != INVALID_COMPANY && (company_id >= MAX_COMPANIES || Company::IsValidID(company_id))) return CMD_ERROR;
-			Company *c = DoStartupNewCompany(true, company_id);
-			if (c != NULL) NetworkServerNewCompany(c, NULL);
+
+			uint num_to_start = max(1U, GB(p2, 0, 4));
```

Comment:
I would say a value of 0 (as a value > `MAX_COMPANIES`) should return `CMD_ERROR`. Not silently be ignored.

Reply:
Uh, that means I have to change p2 everywhere to 1.

Reply:
Sounds you are being lazy :)